rotkehlchen/tests/exchanges/test_ftx.py

```diff
@@ -0,0 +1,585 @@
+
+
+def query_ftx_and_test(
+        ftx: FTX,
+        query_fn_name: str,
+        fills_response: str = FILLS_RESPONSE,
+        deposits_response: str = DEPOSITS_RESPONSE,
+        withdrawals_response: str = WITHDRAWALS_RESPONSE,
+        expected_warnings_num: int = 0,
+        expected_errors_num: int = 0,
+        # Since this test only mocks as breaking only one of the two actions by default
+        expected_actions_num: int = 1,
+):
+    def mock_ftx_query(url):  # pylint: disable=unused-argument
+        if 'fills' in url:
+            return MockResponse(200, fills_response)
+        if 'deposits' in url:
+            return MockResponse(200, deposits_response)
+        if 'withdrawals' in url:
+            return MockResponse(200, withdrawals_response)
+        # else
+        raise AssertionError(f'Unexpected url {url} for test')
+
+    query_fn = getattr(ftx, query_fn_name)
+    with patch.object(ftx.session, 'get', side_effect=mock_ftx_query):
+        actions = query_fn(
+            start_ts=Timestamp(0),
+            end_ts=TEST_END_TS,
+        )
+
+    errors = ftx.msg_aggregator.consume_errors()
+    warnings = ftx.msg_aggregator.consume_warnings()
+    assert len(actions) == expected_actions_num
+    assert len(errors) == expected_errors_num
+    assert len(warnings) == expected_warnings_num
+
+
+def test_ftx_trade_history(mock_ftx):
+    """Test the happy path when querying trades"""
+    with patch.object(mock_ftx.session, 'get', side_effect=mock_normal_ftx_query):
+        trades = mock_ftx.query_online_trade_history(
+            start_ts=0,
+            end_ts=TEST_END_TS,
+        )
+    warnings = mock_ftx.msg_aggregator.consume_warnings()
+    errors = mock_ftx.msg_aggregator.consume_errors()
+    assert len(warnings) == 0
+    assert len(errors) == 0
+    assert len(trades) == 3
+
+    expected_trades = [Trade(
+        timestamp=Timestamp(1614387662),
+        location=Location.FTX,
+        base_asset=A_ETH,
+        quote_asset=A_USD,
+        trade_type=TradeType.SELL,
+        amount=FVal('1'),
+        rate=FVal('2000'),
+        fee=FVal('4'),
+        fee_currency=A_USD,
+        link='1',
+    ), Trade(
+        timestamp=Timestamp(1615338062),
+        location=Location.FTX,
+        base_asset=A_1INCH,
+        quote_asset=A_ETH,
+        trade_type=TradeType.BUY,
+        amount=FVal('400'),
+        rate=FVal('0.005'),
+        fee=FVal('0.0004'),
+        fee_currency=A_ETH,
+        link='2',
+    ), Trade(
+        timestamp=Timestamp(1615424462),
+        location=Location.FTX,
+        base_asset=A_1INCH,
+        quote_asset=A_ETH,
+        trade_type=TradeType.SELL,
+        amount=FVal('200'),
+        rate=FVal('0.006'),
+        fee=FVal('0.6'),
+        fee_currency=Asset('BTC'),
+        link='3',
+    )]
+
+    assert trades == expected_trades
+
+
+def test_ftx_trade_history_unexpected_data(mock_ftx):
+    """Test that the trade history handles invalid data"""
+    query_ftx_and_test(
```

Comment:
oh I see you went and tried to see that all missing data is handled correctly. Well done!

But no need to overkill it with tests this much. At the end of the day we need good enough logging so that we can see what happened when a problem occurs and the user sends us the logs.